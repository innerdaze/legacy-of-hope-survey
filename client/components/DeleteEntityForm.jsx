import React, { PropTypes } from 'react'
import Layer from 'grommet/components/Layer'
import Paragraph from 'grommet/components/Paragraph'
import Footer from 'grommet/components/Footer'
import Anchor from 'grommet/components/Anchor'
import CloseIcon from 'grommet/components/icons/base/Close'
import TrashIcon from 'grommet/components/icons/base/Trash'

const DeleteEntityForm = ({ message, onConfirm, onCancel }) => (
  <Layer>
    <Paragraph
      align='center'>
      {message}
    </Paragraph>
    <Footer
      justify='between'>
      <Anchor
        label='Cancel'
        icon={<CloseIcon/>}
        onClick={onCancel}/>
      <Anchor
        primary={true}
        label='Confirm'
        icon={<TrashIcon/>}
        onClick={onConfirm}/>
    </Footer>
  </Layer>
)

DeleteEntityForm.propTypes = {
  message: PropTypes.string.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired
}

DeleteEntityForm.defaultProps = {
  message: 'Confirm you would like to delete this item',
  onConfirm: Function.prototype,
  onCancel: Function.prototype
}

export default DeleteEntityForm
