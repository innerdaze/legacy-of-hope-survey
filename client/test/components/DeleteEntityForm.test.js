import React from 'react'
import DeleteEntityForm from '../../components/DeleteEntityForm'

let wrapper, message, onConfirm, onCancel

describe('Testing on DeleteEntityForm...', () => {
  beforeEach(() => {
    message = 'Confirm you would like to delete this item'
    onConfirm = jest.fn()
    onCancel = jest.fn()
    wrapper = shallow(
      <DeleteEntityForm
        message={message}
        onConfirm={onConfirm}
        onCancel={onCancel}
      />
    )
  })

  describe('DeleteEntityForm layout', () => {
    test('Expect DeleteEntityForm to have parent Layer comp and child comps Paragraph and Footer', () => {
      expect(wrapper.find('Layer')).toHaveLength(1)
      expect(wrapper.find('Layer').find('Paragraph')).toHaveLength(1)
      expect(wrapper.find('Layer').find('Footer')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Footer comp to have 2 Anchor comps', () => {
      expect(wrapper.find('Footer').find('Anchor')).toHaveLength(2)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect onConfirm to call when Anchor Confirm is clicked', () => {
      wrapper.find('Anchor[label="Confirm"]').simulate('click')

      expect(onConfirm).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect onCancel to call when Anchor Cancel is clicked', () => {
      wrapper.find('Anchor[label="Cancel"]').simulate('click')

      expect(onCancel).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
