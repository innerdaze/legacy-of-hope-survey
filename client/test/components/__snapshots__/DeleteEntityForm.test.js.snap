// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on DeleteEntityForm... DeleteEntityForm layout Expect DeleteEntityForm to have parent Layer comp and child comps Paragraph and Footer 1`] = `
<Layer
  align="center"
>
  <Paragraph
    align="center"
  >
    Confirm you would like to delete this item
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={[MockFunction]}
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <Trash
          responsive={true}
        />
      }
      label="Confirm"
      method="push"
      onClick={[MockFunction]}
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;

exports[`Testing on DeleteEntityForm... DeleteEntityForm layout Expect Footer comp to have 2 Anchor comps 1`] = `
<Layer
  align="center"
>
  <Paragraph
    align="center"
  >
    Confirm you would like to delete this item
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={[MockFunction]}
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <Trash
          responsive={true}
        />
      }
      label="Confirm"
      method="push"
      onClick={[MockFunction]}
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;

exports[`Testing on DeleteEntityForm... Test on prop Expect onCancel to call when Anchor Cancel is clicked 1`] = `
<Layer
  align="center"
>
  <Paragraph
    align="center"
  >
    Confirm you would like to delete this item
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <Trash
          responsive={true}
        />
      }
      label="Confirm"
      method="push"
      onClick={[MockFunction]}
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;

exports[`Testing on DeleteEntityForm... Test on prop Expect onConfirm to call when Anchor Confirm is clicked 1`] = `
<Layer
  align="center"
>
  <Paragraph
    align="center"
  >
    Confirm you would like to delete this item
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={[MockFunction]}
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <Trash
          responsive={true}
        />
      }
      label="Confirm"
      method="push"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;
